From gtk-2-14 branch:

2008-09-25  Michael Natterer

	gtk/gtkfilechooserdefault.c (gtk_file_chooser_default_finalize):
	don't unref the file system backend, the newly added
	unset_file_system_backend() already does this (bug #553135).

--- branches/gtk-2-14/gtk/gtkfilechooserdefault.c	2008/09/25 11:22:34	21514
+++ branches/gtk-2-14/gtk/gtkfilechooserdefault.c	2008/09/25 11:23:42	21515
@@ -934,8 +934,6 @@
 
   shortcuts_free (impl);
 
-  g_object_unref (impl->file_system);
-
   g_free (impl->browse_files_last_selected_name);
 
   for (l = impl->filters; l; l = l->next)
